src/main/download.py

```diff
@@ -12,20 +12,28 @@ def download_model(path, model):
     Otherwise, the model should exist on disk.
     @param path: Path to where the models will be downloaded.
     @param model: Model object.
-    @return: Path to the model on disk.
+    @return: Path to the model on disk and if the model already exists
+    @raise: throws an error if the model can't be downloaded and remove dir
     """
     model_name = model['model']
     model_path = "{}/{}/".format(path, model_name)
-    if not os.path.isdir(model_path):
-        os.makedirs(model_path)  # Create the dir
-        print("Downloading model {} to {}".format(model_name, model_path))
-        download_from_huggingface(model_name, model_path)
-    else:
-        print("Model already exists: {}".format(model_path))
-    return model_path
+    model_exists = False
+    try:
+        if not os.path.isdir(model_path):
+            os.makedirs(model_path)  # Create the dir
+            print("Downloading model {} to {}".format(model_name, model_path))
+            download_from_huggingface_qa(model_name, model_path)
```

Comment:
What happens if the model is for sentence transformers? Does this download method only supports question and answer models?

Reply:
At that time the sentence transformer model was not in the scope, to see the latest version and sentence transformer implementation please check next PR